pandas/tests/series/test_constructors.py

```diff
@@ -79,6 +79,23 @@ def test_infer_with_date_and_datetime(self):
         expected = Index(vals, dtype=object)
         tm.assert_index_equal(idx, expected)
 
+    @pytest.mark.parametrize("dtype", ["M8[s]", "M8[ms]", "M8[us]", "M8[ns]"])
+    def test_constructor_str_object_dtypes_dt64_array(self, dtype):
+        # GH 57512
+        dt_arr = np.array(
+            [
+                "2024-01-03T00:00:00.000000000",
+                "2024-01-01T00:00:00.000000000",
+            ],
+            dtype=dtype,
+        )
+        result = Series(Series(dt_arr, dtype=str), dtype=dtype)
```

Comment:
I am a little unsure about this change and the issue from the OP - what the is reason for trying to have these losslessly convert between strings and datetimes? If you remove the string conversions things work fine right?

Reply:
Thanks for the feeback and apologies for the delayed response.

While investigating the original problem, I tried to find the origin of the timestamps we see in the OP, and I found out that the odd read_csv behaviour started because of some refactoring that was done it.

I narrowed it down to it being the Series constructor in this refactored code that ends up generating the timestamps when called with a dt64[ns] nparray and dtype=str:
https://github.com/pandas-dev/pandas/blob/fb05cc7781637906728e95be9b418e774b0e67b2/pandas/io/parsers/readers.py#L1872-L1880

I thought this behaviour from the Serier ctor was not intended since it seems to be isolated and stands out from other combinations of types:

For example, doing the exact same ctor call but with a dt64 nparray that has any precision level other than nanoseconds works fine, and the resulting Series is able to be losslessly converted back to a dt64 Series (without the fix, the test case I added already passes whenever it runs with an array of an M8 type that is **not** the nanosecond one).
The same behaviour can be observed when calling it with a dt64[ns] nparray and dtype=object instead of str.

This seemed like the standard and it made sense to me that we could recover the dates from the resulting strings.

I thought the best solution to this would be to deal with the root cause (assuming it's unintended behaviour) instead of trying to patch read_csv to work as before.

I'm not 100% sure if I understood what you meant by removing the string conversions, however, if we do not call the ctor with dtype=str, everything works as expected since the problem only arises when we try to convert dt64[ns] dates in an nparray to string dtype using the Series ctor.

Reply:
> I'm not 100% sure if I understood what you meant by removing the string conversions, however, if we do not call the ctor with dtype=str, everything works as expected since the problem only arises when we try to convert dt64[ns] dates in an nparray to string dtype using the Series ctor.

Yea ultimately I am trying to figure out what a user is expecting to have happen when specifying `dtype=str` alongside `parse_dates=` - are they trying to get back a date or a string? I understand this may have worked previously but on the flip side if its ambiguous what to expect we _may_ decide we want to warn / deprecate when receiving conflicting arguments like htat

Reply:
Ah I see, and I totally agree, I feel like it would be better if the user was at least warned about the incompatible arguments they specified.
But I'd be interested in hearing your thoughts on the Series ctor behaviour by itself, since, to me, it seems a bit unintuitive how it treats M8[ns] arrays differently from other M8 type arrays.